.github/matchers/gcc-problem-matcher.json

```diff
@@ -0,0 +1,17 @@
+{
+    "problemMatcher": [
+        {
+            "owner": "gcc-problem-matcher",
+            "pattern": [
+                {
+                    "regexp": "^(.*?):(\\d+):(\\d*):?\\s+(?:fatal\\s+)?(warning|error):\\s+(.*)$",
+                    "file": 1,
```

Comment:
What is the reason for the first expression: `(.*?)`?

Reply:
The first expression captures the filename.

Reply:
Why this form :-)